app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java

```diff
@@ -1564,41 +1569,29 @@ public void draftEmail(final Context context, final String subject, final String
      */
     @SuppressWarnings({"ResultOfMethodCallIgnored", "ConstantConditions"})
     public File extractFileFromIntent(final Context context, final Intent receivingIntent) {
-        String action = receivingIntent.getAction();
-        String type = receivingIntent.getType();
+        final String action = receivingIntent.getAction();
+        final String type = receivingIntent.getType();
+        final String extPath = Environment.getExternalStorageDirectory().getAbsolutePath();
         String tmps;
         String fileStr;
-        String[] sarr;
-        final ArrayList<String> probeFiles = new ArrayList<>();
-
-        // Filter non existing files out
-        GsCallback.a0 filterNe = () -> {
-            for (String fp : new ArrayList<>(probeFiles)) {
-                boolean ok = false;
-                if (!TextUtils.isEmpty(fp)) {
-                    File f = new File(fp);
-                    ok = f.exists() && f.canRead();
-                }
-                if (!ok) {
-                    probeFiles.remove(fp);
-                }
-            }
-        };
+        File result = null;
 
         if ((Intent.ACTION_VIEW.equals(action) || Intent.ACTION_EDIT.equals(action)) || Intent.ACTION_SEND.equals(action)) {
+
             // Màrkor, SimpleMobileTools FileManager
             if (receivingIntent.hasExtra((tmps = EXTRA_FILEPATH))) {
-                probeFiles.add(receivingIntent.getStringExtra(tmps));
+                result = checkPath(receivingIntent.getStringExtra(tmps));
```

Comment:
Instead of adding all the test paths to a list and then filtering it, we just test each time. The subsequent tests don't happen if a previous test passes. This prevents additional work and cleans up a couple of cases.

Reply:
Need to try this with lots of cases and apps. Hope it not breaks